### ISSUE TITLE
Brave 1.63.x does not apply certain(?) fonts listed in site's CSS. This is a regression from 1.62.x

### ISSUE THREAD

**Issue by m0nkey-br4in** [brave/brave-browser#36389]
## Description 

Brave 1.63.x does not apply certain(?) fonts listed in site's CSS. 
I regularly visit, say, https://www.old-games.ru/forum/. It has the following stanza in its main css:
```css
body: {
    font-family: 'Trebuchet MS', Helvetica, Arial, sans-serif;
} 
```
I have Trebuchet MS installed locally. Despite this all, text is rendered with PT Sans (if Stylus extension is active) or Segoe UI (if it's inactive).
Everything was correct in Brave 1.62.165

**Disabling Brave Shields fixes the problem**

## Steps to Reproduce

   1.  Visit a site you frequent
   2.  Wonder why it looks somewhat off

## Actual result:

On 1.63.162:
![Screenshot from 2024-02-27 13-23-53](https://github.com/brave/brave-browser/assets/8735883/ff9f8a7c-cff6-41e3-bcd2-f4b955297af9)
Notice "font-family" and "rendered fonts" in the inspector panel

## Expected result:
On 1.62.165:
![Screenshot from 2024-02-27 13-21-48](https://github.com/brave/brave-browser/assets/8735883/dd8d54d3-82bd-4bec-bbe6-54b976fc4d5b)

## Reproduces how often: 

Reproduces always

## Brave version (brave://version info)

Brave:	1.63.162 Chromium: 122.0.6261.69 (Official Build) (64-bit) 
Revision:	3098e6bd850eaf3aa13541d2ddd87af53908bc60
OS:	Linux (specifically, Ubuntu 22.04)

## Version/Channel Information:

Reproduces on current release channel. Did not check beta/nightly

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? 
  Yes, disabling Brave Shields fixes the problem
- Does the issue resolve itself when disabling Brave Rewards?
  Don't know how to enable/disable them. Probably have them disabled, right? 
- Is the issue reproducible on the latest version of Chrome? 
  Don't have Chrome, but in Chromium version 122.0.6261.69 (Official Build) snap (64-bit) works correctly

**Comment by rebron** [comment]
cc: @ShivanKaul

**Comment by ShivanKaul** [comment]
Hmm wonder if it's a Linux-only issue. Can you confirm if this happens with Fingeprinting set to be both `Standard` and `Strict`? Or is it only happening on `Strict`? 1.63.x still has `Strict` mode.

**Comment by flaudisio** [comment]
Hi, Shivan. I've tested again and it happens with Fingerprinting set to both `Standard` and `Strict`; only `Disabled` fixes it for now.

**Comment by ShivanKaul** [comment]
Gotcha. Another test: does it get fixed if you disable "Prevent sites from fingerprinting me based on my language preferences" global option in brave://settings/shields?

<img width="1119" alt="image" src="https://github.com/brave/brave-browser/assets/5284154/ee078d3b-0679-4068-a196-679c25e4dc44">

**Comment by flaudisio** [comment]
Yes! It's fixed after disabling it, even with _Block fingerprinting_ changed back to `Standard`.

**Comment by ShivanKaul** [comment]
The interim fix is to disable "Prevent sites from fingerprinting me based on my language preferences" global option in brave://settings/shields but we'll address this soon.

**Comment by arthuredelstein** [comment]
Hi @jknockel -- great analysis, thank you! Sounds like have a couple of possible options:
1. Add the font aliases to the allowlist, as you suggest, or
2. Make sure aliases are applied *before* we filter fonts through the allowlist.

**Comment by doebi** [comment]
> Same here. I've noticed the default font in Google Docs was broken, falling back to the default serif font. Fixed when I disabled Shield's _Block fingerprinting_ option.
> 
> Before disabling _Block fingerprinting_:
> 
> ![brave-01](https://private-user-images.githubusercontent.com/285751/308578306-91af3df1-0c8f-42b9-9be4-0569f27e38bd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI3NDY3MzUsIm5iZiI6MTcxMjc0NjQzNSwicGF0aCI6Ii8yODU3NTEvMzA4NTc4MzA2LTkxYWYzZGYxLTBjOGYtNDJiOS05YmU0LTA1NjlmMjdlMzhiZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQxMFQxMDUzNTVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05NDIwMWZmOTkwMzI4YjNmMTc0MjFjMGMxY2I2OWEyZjkxMjVhOGM1NjRjMTg2NjI3NmZhZDFkMmExOGQ1MDIyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.4XRj5OOOf9j6JLxnu0B4gNUlfO0mhgo50829asccyP4)
> 
> After disabling it and reloading the page:
> 
> ![brave-02](https://private-user-images.githubusercontent.com/285751/308578523-596e6e28-cb63-4ef1-a379-611c8486f1fd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI3NDY3MzUsIm5iZiI6MTcxMjc0NjQzNSwicGF0aCI6Ii8yODU3NTEvMzA4NTc4NTIzLTU5NmU2ZTI4LWNiNjMtNGVmMS1hMzc5LTYxMWM4NDg2ZjFmZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQxMFQxMDUzNTVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lZDdiYmI1NzRmZTU3ODRkNDM5NmQ1YjJjNmM5NTAyODkwNmMzMDc2YzMzZWJmZTNkNmUzYjQwOWRmOThiMjM3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.6IVvWrAPIAw8Fs2O730AET6kJUs-o-5haOKr7e0sM6s)
> 
> **Brave version (brave://version info):**
> 
> Brave: 1.63.162 Chromium: 122.0.6261.69 (Official Build) (64-bit) Revision: 3098e6bd850eaf3aa13541d2ddd87af53908bc60 OS: Linux (Xubuntu 22.04)

I want to raise awareness to this circumstance again, as this is affecting essentially any website without an explicitly set font. When will this eventually be fixed?

**Comment by Rawi666** [comment]
I'd like to add more to this problem. There are multiple sites that are broken on default installation of Fedora 42 because they are using 'Open Sans' fonts. Example: https://benchmark.pl.
This site is using 'Open Sans' font
- on default installation of Ubuntu 25.04 with Brave the site is looking good
- on default installation of Fedora 42 Brave is using 'sans' font which breaks user experience.
Yes, disabling "Prevent sites from fingerprinting me based on my language preferences" solves the problem but it is extremely difficult to find out that this setting solves it.
